contrib/inventory/ec2.py

```
@@ -672,7 +672,9 @@ def get_elasticache_clusters_by_region(self, region):
                 error = self.get_auth_error_message()
             if not e.reason == "Forbidden":
                 error = "Looks like AWS ElastiCache is down:\n%s" % e.message
-            self.fail_with_error(error, 'getting ElastiCache clusters')
+                self.fail_with_error(error, 'getting ElastiCache clusters')
+            else:
+                return
```

this swallows the exception from the user's perspective.  maybe:
```else: 
  raise
```

applies to the next change down too

---

This is precisely what I'm trying to avoid. It should be transparent to me - if I don't have permissions, I don't actually care, I don't want it to error out, I'm happy for the Elasticache results to be empty.

---

I consider that a violation of least surprise.  if you have your ec2_inventory set to get elasticache or rds instances and it returns none due to error when in fact there are instances, that is surprising.

---

If I have a user with no RDS permissions, for example, and no RDS instances, I consider it a gross violation of least surprise that ec2.py errors out rather than working as expected.

---

disagree.  it is an error to request something you don't have rights for.  you don't know that there aren't any rds instances.
i'd suggest that addressing the situation you're discussing would be better done defaulting `include_rds_clusters` to false in the default ini file.

---

I don't know, but neither do I care, if I don't have the perms, whether there are any RDS instances.

The default ec2.ini should be suitable for most purposes, and not error in the legitimate case that a locked down user exists.